x/subscription/types/subscription.go

```diff
@@ -0,0 +1,33 @@
+package types
+
+import (
+	"strings"
+	"time"
+
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+func (sub Subscription) IsExpired(date time.Time) bool {
+	expiry := time.Unix(int64(sub.ExpiryTime), 0).UTC()
+	return expiry.Before(date)
+}
+
+// ValidateSubscription validates a subscription object fields
+func (sub Subscription) ValidateSubscription() error {
```

Comment:
isn't this a duplicate of your ValidateBasic() function?

Reply:
validate basic is done for messages and proposals